chia/wallet/wallet.py

```diff
@@ -276,16 +276,70 @@ async def select_coins(self, amount, exclude: List[Coin] = None) -> Set[Coin]:
         unconfirmed_removals: Dict[bytes32, Coin] = await self.wallet_state_manager.unconfirmed_removals_for_wallet(
             self.id()
         )
+        # check for exact 1 to 1 coin match.
         for coinrecord in unspent:
-            if sum_value >= amount and len(used_coins) > 0:
-                break
             if coinrecord.coin.name() in unconfirmed_removals:
                 continue
             if coinrecord.coin in exclude:
                 continue
-            sum_value += coinrecord.coin.amount
-            used_coins.add(coinrecord.coin)
-            self.log.debug(f"Selected coin: {coinrecord.coin.name()} at height {coinrecord.confirmed_block_height}!")
+            if coinrecord.coin.amount == amount:
+                sum_value += coinrecord.coin.amount
+                used_coins.add(coinrecord.coin)
+                self.log.debug(
+                    f"Selected coin: {coinrecord.coin.name()} at height {coinrecord.confirmed_block_height}!"
+                )
+                break
+
+        # Check for exact match with all coins smaller than the amount.
+        # If the total of all the smaller coins are less than the amount we choose; the smallest coin available.
```

Comment:
Fix grmmar here, remove semicolon. Choose the smallest coin available that is still greater than the amount correct? What if there are no coins big enough, do we fail here?